tests/registration_test.py

```diff
@@ -533,6 +533,114 @@ def test_strings_groupby_attach(self):
         self.assertIsInstance(sAttach.segments, ak.pdarray)
         self.assertIsInstance(sAttach.unique_keys, ak.Strings)
 
+    def test_string_property_accessor(self):
+        cleanup()
+        s = ak.array(["one", "two", "three", "123"])
+        s.register("sname")
+        ak.clear()
+        sAttach = ak.Strings.attach("sname")
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        s.get_bytes()
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        self.assertListEqual(s.get_bytes().to_list(), sAttach.get_bytes().to_list())
+        self.assertListEqual(s.get_offsets().to_list(), sAttach.get_offsets().to_list())
+        s.get_offsets()
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        self.assertListEqual(s.get_bytes().to_list(), sAttach.get_bytes().to_list())
+        self.assertListEqual(s.get_offsets().to_list(), sAttach.get_offsets().to_list())
+
+        cleanup()
+        s = ak.array(["one", "two", "three", "123"])
+        s.register("sname")
+        ak.clear()
+        s.get_bytes()
+        s.get_offsets()
+        sAttach = ak.Strings.attach("sname")
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        self.assertListEqual(s.get_bytes().to_list(), sAttach.get_bytes().to_list())
+        self.assertListEqual(s.get_offsets().to_list(), sAttach.get_offsets().to_list())
+
+        cleanup()
+        s = ak.array(["one", "two", "three", "123"])
+        s.register("sname")
+        ak.clear()
+        s.get_offsets()
+        sAttach = ak.Strings.attach("sname")
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        self.assertListEqual(s.get_bytes().to_list(), sAttach.get_bytes().to_list())
+        self.assertListEqual(s.get_offsets().to_list(), sAttach.get_offsets().to_list())
+        s.get_bytes()
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        self.assertListEqual(s.get_bytes().to_list(), sAttach.get_bytes().to_list())
+        self.assertListEqual(s.get_offsets().to_list(), sAttach.get_offsets().to_list())
```

Comment:
Again, I think this block effectively tests the same thing as the previous 2. If we want to maintain it to ensure that the calling order does not present any issues, I would suggest updating to the format from my previous comment.